webdriver/get-title.py

```diff
@@ -0,0 +1,268 @@
+
+# [...]
+# 2. Handle any user prompts and return its value if it is an error.
+# [...]
+# In order to handle any user prompts a remote end must take the following
+# steps:
+# 2. Run the substeps of the first matching user prompt handler:
+#
+#    [...]
+#    - missing value default state
+#    - not in the table of simple dialogs
+#      1. Dismiss the current user prompt.
+#      2. Return error with error code unexpected alert open.
+def test_title_handle_prompt_missing_value(session, create_dialog):
+    document = inline("<title>This is the document's title</title>")
+    session.start()
+    session.send_session_command("POST", "url", dict(url=document))
+    assert_alert_handled = create_dialog("alert")
+
+    result = session.transport.send("GET",
+                                    "session/%s/title" % session.session_id)
+
+    assert_error(result, "unexpected alert open")
+    assert_alert_handled(None)
+
+    assert_confirm_handled = create_dialog("confirm")
+
+    result = session.transport.send("GET",
+                                    "session/%s/title" % session.session_id)
+
+    assert_error(result, "unexpected alert open")
+    assert_confirm_handled(False)
+
+    assert_prompt_handled = create_dialog("prompt")
+
+    result = session.transport.send("GET",
+                                    "session/%s/title" % session.session_id)
+
+    assert_error(result, "unexpected alert open")
+    assert_prompt_handled(None)
+
+# The behavior of the `window.print` function is platform-dependent and may not
+# trigger the creation of a dialog at all. Therefore, this test should only be
+# run in contexts that support the dialog (a condition that may not be
+# determined automatically).
+#def test_title_with_non_simple_dialog(session):
+#    document = "<title>With non-simple dialog</title><h2>Hello</h2>"
+#    spawn = """
+#        var done = arguments[0];
+#        setTimeout(function() {
+#            done();
+#        }, 0);
+#        setTimeout(function() {
+#            window.print();
+#        }, 0);
+#    """
+#    session.url = inline(document)
+#    session.execute_async_script(spawn)
+#
+#    result = session.transport.send("GET",
+#                                    "session/%s/title" % session.session_id)
+#    assert_error(result, "unexpected alert open")
+
+# [...]
+# 3. Let title be the initial value of the title IDL attribute of the current
+#    top-level browsing context's active document.
+# 4. Return success with data title.
+# [...]
+# The title attribute must, on getting, run the following algorithm:
+# [...]
+# 2. Otherwise, let value be the child text content of the title element [...]
+# [...]
+# 4. Return value.
+def test_title_from_top_context(session):
+    session.url = inline("<title>Foobar</title><h2>Hello</h2>")
+
+    result = session.transport.send("GET",
+                                    "session/%s/title" % session.session_id)
+    assert_success(result, "Foobar")
```

Comment:
Can you elaborate? I'm interpreting your comment to mean an assertion like:

    assert client.execute_script('return document.title;') == 'Foobar'

...but that looks like a test for [the `title` attribute in HTML](https://html.spec.whatwg.org/#document.title) rather than a WebDriver test.